openlcs/libs/sc_handler.py

```diff
@@ -166,11 +183,19 @@ def get_source_container_srpms_metadata(self):
         """
         misc_dir = os.path.join(self.dest_dir, 'metadata')
         misc_dir = create_dir(misc_dir)
-        patterns = ['/*.json', '/repositories']
-        for pattern in patterns:
-            misc_files = glob.glob(self.dest_dir + pattern)
-            for file in misc_files:
-                shutil.move(file, misc_dir)
+
+        # Remove all the "layer.tar" files, because they are unuseful
+        # soft link files.
+        layer_file_paths = glob.glob(self.dest_dir + '/**/layer.tar')
+        for layer_file_path in layer_file_paths:
+            os.remove(layer_file_path)
+
+        # Move all the misc files, directories to "metadata" directory.
+        nested_items = os.listdir(self.dest_dir)
+        for nested_item in nested_items:
+            if nested_item not in ['extra_src_dir', 'rpm_dir', 'metadata']:
+                item_path = os.path.join(self.dest_dir, nested_item)
+                shutil.move(item_path, misc_dir)
```

Comment:
I am not sure if we got the agreement of moving other files to the misc_dir. If I remember correctly, there are many files in the vendor directory of golang container. Do we need to scan all the files?

Reply:
yes, all the files that not belong component should move to misc, remote source move at other function, not here